File: solutions/c/resistor-color-trio/1/resistor_color_trio.c
#include "resistor_color_trio.h"

resistor_value_t color_code(resistor_band_t *resist){
    resistor_value_t res;
    resistor_band_t res_arr[3]={BLACK, BLACK, BLACK};
    unsigned long i;
    res_arr[0]=*resist;
    resist++;
    res_arr[1]=*resist;
    resist++;
    res_arr[2]=*resist;
    resist++;
    if(res_arr[2]<2){
        res.unit=OHMS;
    }
    else if(res_arr[2]<5){
        res.unit=KILOOHMS;
    }
    else if(res_arr[2]<8){
        res.unit=MEGAOHMS;
    }    
    else if(res_arr[2]<=9){
        res.unit=GIGAOHMS;
    }
    res.value=10*res_arr[0]+res_arr[1];
    for(i=0;i<res_arr[2];i++){
        res.value=res.value*10;
    }
    if(res.unit==KILOOHMS){
        res.value=res.value/1000;
    }
    else if(res.unit==MEGAOHMS){
        res.value=res.value/1000000;    
    }
    else if(res.unit==GIGAOHMS){
        res.value=res.value/1000000000;    
    }
    return res;
}

File: solutions/python/resistor-color-trio/1/resistor_color_trio.py
def color_code(color):
    return colors().index(color)

def colors():
    color_list=[
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
        ]
    return color_list

def value(colors):
    resistor=""
    for color in colors:
        resistor=resistor+str(color_code(color))

    return int(resistor[:2])

def zeros(color):
    zero=''
    for i in range(0,color_code(color)):
        zero=zero+'0'

    return zero

def label(colors):
    resistor=str(value(colors[0:2]))+zeros(colors[2])
    res_value=''
    res_mult=''
    res_zeros=resistor.count('0')
    if res_zeros < 3:
        res_mult='ohms'
        res_value=resistor
    elif res_zeros < 6:
        res_mult='kiloohms'
        res_value=resistor[0:len(resistor)-3]
    elif res_zeros < 9:
        res_mult='megaohms'
        res_value=resistor[0:len(resistor)-6]
    else:
        res_mult='gigaohms'
        res_value=resistor[0:len(resistor)-9]
    return res_value+' '+res_mult
    


